### DISCUSSION TITLE
[BUG] Certificate revocation error - "Search End Entity" page does not reload the status correctly

### DISCUSSION THREAD

**Discussion by DanieleSmith92** [Keyfactor/ejbca-ce#741] [General]
I have a local installation of EJBCA in a local VM. It was running version 8.3.2 correctly on wildfly-26.1.3 with java 17.0.13.
 
On the "Search End Entity" page you have the possibility to revoke the certificate via the "Revoke Selected" button 
but also by clicking on the "Certificate" button, present in line with the certificate under the "Actions" column, and then click on "View". 

<img width="1440" alt="1" src="https://github.com/user-attachments/assets/8deec3f0-3ba5-43b9-93c3-e8783a759573" />

By clicking on "Certificate" -> "View", the modal opens. Into the modal, by selecting the reason for revocation we can revoke the certificate by clicking on "Revoke". 

<img width="1440" alt="2" src="https://github.com/user-attachments/assets/325ff206-ef14-47ad-adaa-9814cce2a50d" />

the result is

<img width="1440" alt="3" src="https://github.com/user-attachments/assets/2d6580d4-ed30-47d9-8a5b-4c30c46a9f1c" />

After closing the modal and refreshing the "Search End Entity" page, the "Status" column is not updated and remains in "GENERATED".

<img width="1440" alt="4" src="https://github.com/user-attachments/assets/6142c01d-66d2-4391-ba87-51d04ec4dff9" />

Instead, performing the same operation from the "Search End Entity" page, after selecting a new test(test11) via the checkbox, by clicking on "Revoke Selected", the "Status" column is correctly updated to "REVOKED".

**Comment by primetomas** [comment]
The difference is End Entities vs Certificates. One end entity may have multiple certificates. You can revoke one, or several of an end entities certificates. This does not make the End Entity revoked. When you "select and revoke" in the end entity page it ask you "are you sure you want to revoke this end entity", i.e. not the same as certificate.
Revoking an end entity makes it impossible to renew a certificate with a protocol that uses an old certificate to authenticate/authorize renewal. While revoking an individual certificate does not prevent the end entity from using, and renewing, another non-revoked certificate.

Does it sound complicated, maybe yes :-)
